templates/nextjs/src/pages/predicate.tsx

```diff
@@ -0,0 +1,150 @@
+import { Button } from "@/components/Button";
+import { FuelLogo } from "@/components/FuelLogo";
+import { Input } from "@/components/Input";
+import { Link } from "@/components/Link";
+import { TestPredicateAbi__factory } from "@/sway-api/predicates/index";
+import type { BN, InputValue, Predicate, WalletUnlocked } from "fuels";
+import { BaseAssetId, Provider, Wallet, bn } from "fuels";
+import { useEffect, useState } from "react";
+
+export default function PredicateExample() {
+  const [predicate, setPredicate] = useState<Predicate<InputValue[]>>();
+  const [wallet, setWallet] = useState<WalletUnlocked>();
+
+  const [walletBalance, setWalletBalance] = useState<BN>();
+  const [predicateBalance, setPredicateBalance] = useState<BN>();
+
+  const [pin, setPin] = useState<string>();
+
+  useEffect(() => {
+    (async () => {
+      const provider = await Provider.create("http://127.0.0.1:4000/graphql");
+
+      // 0x1 is the private key of one of the fauceted accounts on your local Fuel node
+      const wallet = Wallet.fromPrivateKey("0x01", provider);
```

Comment:
I wonder if this is clear enough or how to add more clarity around default faucets.

Perhaps we could generate projects with a `chainConfig.json` by default to reduce the abstractions, making things more explicit for users to customize. It could be side-by-side with the `fuels.config.ts` or something.

Reply:
Interesting. Or maybe instead of the whole chain config, we could start with adding some of these options to the `fuels.config.ts` file itself?

Reply:
Maybe we can turn this into a discussion

Reply:
The `fuels.config.ts` already have the [chainConfig](https://fuellabs.github.io/fuels-ts/guide/cli/config-file.html#chainconfig) property, and it's the path for the file. 

I'd say we should keep `chainConfig.json` as a JSON file for consistency and portability.

Users can use it for spinning up `fuel-core` nodes using their binaries instead of the [built-in](https://fuellabs.github.io/fuels-ts/guide/cli/built-in-binaries.html#built-in-binaries) ones.

Reply:
@Dhaiwat10 Feel free to open the discussion with your proposal.